### ISSUE TITLE
Feature request:  Highlight roads missing maxspeed or maxspeed:* values

### ISSUE THREAD

**Issue by jfburke3** [osmandapp/Osmand#8239]
A "nice to have" feature request:  A setting or plugin that will highlight roads missing maxspeed=* or maxspeed:* (e.g., maxspeed:advisory=*) values, similar to how the OSM Mapper Assistant setting highlights FIXME tags.  It could even simply be another function activated by that setting.

Rationale:  Some people who participate in Mapillary or OpenStreetCam (including myself) might be willing to take a detour through a particular area if it was obvious that speed limits were missing on some of the roads.

As I've conceived it, a road that has a valid maxspeed=* value would not be highlighted at all.  A road that is missing maxspeed=* but has a valid maxspeed:*=* _might_ be highlighted in a subtle color (if at all--up for discussion).  A road that is missing valid values for both maxspeed=* _and_ maxspeed:*=* would be highlighted more prominently.

Only values that _include numbers_, or where the value is "variable" or "none", should be valid.  I've come across some roads where people have tagged maxspeed=unsigned or maxspeed=missing, which should obviously be highlighted as if the value was missing.

**Comment by vshcherb** [comment]
Yeah we need to think about more generic approach how to highlight ways and other objects depends on certain filter conditions. I would definitely say that this feature will come after selecting and highlighting routes.

**Comment by NotSoImportant** [comment]
You can use the StreetComplete app for the purpose of adding missing speed limits.

More complexe question though is how to identify and update incorrect speed limits. There have been several discussions here and in StreetComplete, why this is not trivial.

**Comment by pebogufi** [comment]
Getting an overview about missing or existing speed limits works with (change parameters)
http://mijndev.openstreetmap.nl/~peewee32/maxspeed/Maxspeed.htm?map=surface&zoom=13&lat=48.07036&lon=7.81139&layers=0B000TTTTTTTTT

**Comment by Lee-Carre** [comment]
Further to @pebogufi's suggestion, anyone may use (the more generic) [Overpass Turbo](//www.overpass-turbo.eu/) to find exactly what they want, in their area of interest.

For example, via the wizard

    (type:way & highway=* & maxspeed!=*) in [name of your area]

else omit “in [name of your area]” to match within the bbox of the map view. Read the documentation for more.

This will find all highways (of any type) without a maxspeed set.

Adjust as needed.

---

Let's not reinvent the wheel; use the right tool for the right job.

**Comment by jfburke3** [comment]
@NotSoImportant @pebogufi and @Lee-Carre You're missing my point.  I know about those tools.  I just can't use them when I'm driving.  What I'm interested in is a setting in OsmAnd that will highlight roads as I drive through an area, so that I know they need to be surveyed, so I can survey them as I have time.  OsmAnd already does this for FIXME tags on items, and sometimes I have been able to drive through an area to get the information described by it.

**Comment by NotSoImportant** [comment]
This sounds a bit contradicting to me. When you say you cannot use them while driving, as well such highlighting will not help you.
When however you are able to "drive through an area to get the information described by it" that is exactly the usecase of StreetComplete.
How would you currently "get" or add the information for such fixmes while driving? Probably by stopping when you see something. Same you can do with StreetComplete and the speedlimit quest.

**Comment by jfburke3** [comment]
@NotSoImportant Well, as I mentioned that I participate in bothh Mapillary and KartaView (OpenStreetCam), presumably I would record imagery of the streets in question, upload them to their respective services, and then later use the uploaded imagery to update  OSM.

Or, I could even  just use OsmAnd's built-in a/v notes feature to record short video that includes the speed limit signs, or an audio note to myself that "Caribou Road has speed limit 35 mph" and then update OSM using that info.

StreetComplete is nice, but unfortunately impossible to use _while driving a car_ because it requires a considerable amount of interaction and manipulation using my hands, which are otherwise occupied with the aforementioned car-driving.  (Fun Fact:  in many states in the U.S., it would actually be illegal to use StreetComplete unless the vehicle you are driving is comnpletely parked, because of how much interaction and typing is necessary to use that app in any meaningful way.)
